account_invoice_refund_code/views/account_journal_views.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_account_journal_form" model="ir.ui.view">
+        <field name="name">account.journal.form - account_invoice_refund_code</field>
+        <field name="model">account.journal</field>
+        <field name="inherit_id" ref="account.view_account_journal_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='code']" position="after">
+                <field name="refund_code" />
```

Comment:
we could make the field invisible if you do not choose to have a different sequence for refunds. Also we should hide the field in the same way it is done for the refund sequence (`attrs="{'invisible': [('type', 'not in', ['sale', 'purchase'])]}"`)